internal/provider/providers/myaddr/provider.go

```diff
@@ -0,0 +1,178 @@
+package myaddr
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/netip"
+	"net/url"
+	"strings"
+
+	"github.com/qdm12/ddns-updater/internal/models"
+	"github.com/qdm12/ddns-updater/internal/provider/constants"
+	"github.com/qdm12/ddns-updater/internal/provider/errors"
+	"github.com/qdm12/ddns-updater/internal/provider/headers"
+	"github.com/qdm12/ddns-updater/internal/provider/utils"
+	"github.com/qdm12/ddns-updater/pkg/publicip/ipversion"
+)
+
+type Provider struct {
+	key        string
+	name       string
+	ipVersion  ipversion.IPVersion
+	ipv6Suffix netip.Prefix
+}
+
+func New(data json.RawMessage, ipVersion ipversion.IPVersion, ipv6Suffix netip.Prefix) (p *Provider, err error) {
+	var providerSpecificSettings struct {
+		Key string `json:"key"`
+	}
+	if err = json.Unmarshal(data, &providerSpecificSettings); err != nil {
+		err = fmt.Errorf("json decoding provider specific settings (myaddr): %w", err)
+		return
```

Comment:
```suggestion
		return nil, fmt.Errorf("json decoding provider specific settings: %w", err)
```

Also no point having the provider name, that's wrapped by calling layers (or should be, if not!)